packages/cf-component-link/src/Link.js

```diff
@@ -47,22 +48,43 @@ class Link extends React.Component {
       }
     }
 
-    props.className = 'cf-link';
-
     if (disabled) {
-      props.className += ' cf-link--disabled';
       props.disabled = true;
     }
 
-    if (className) {
-      props.className += ' ' + className;
-    }
-
     props.onClick = this.handleClick;
 
     props.ref = node => this.link = node;
 
-    return React.createElement(tagName, props, children);
+    const Link = createComponent(
+      ({ theme, disabled }) => ({
+        color: disabled ? theme.disabledColor : theme.color,
+        outline: theme.outline,
+        textDecoration: theme.textDecoration,
+        transition: theme.transition,
+        cursor: disabled ? theme.disabledCursor : theme.cursor,
+
+        '&:hover': {
+          color: disabled
+            ? theme['&:hover'].disabledColor
+            : theme['&:hover'].color
+        },
+
+        '&:focus': {
+          color: theme['&:focus'].color,
+          outline: theme['&:focus'].outline,
+          outlineOffset: theme['&:focus'].outlineOffset
+        },
+
+        '&:active': {
+          color: theme['&:active'].color,
+          outline: theme['&:active'].outline
+        }
+      }),
+      tagName,
```

Comment:
The component didn't do any kind of validation of the `tagName` before so I'm unsure what these would be. I'd prefer just to have the `a` tag here and not allow it to be customized since I can't see the use case for it.